### PR TITLE
[SPARK-35328][Core] Use 'SPARK_DRIVER_LOG_URL_' as env prefix for getting driver log urls by default

### PR DESCRIPTION
**Pull request by sharkdtu**
### What changes were proposed in this pull request?
Currently, spark on kubernetes can't show logs url on ui. To check history logs, we usually collect pod logs to third-party logging services, which can be accessed by urls. To show log urls on ui,  we can set env prefixed with 'SPARK_LOG_URL_' for executors. But for driver, there is no way to show log urls by setting env. 

This pr use 'SPARK_DRIVER_LOG_URL_' as env prefix for getting driver log urls by default.

### Why are the changes needed?

To show driver log urls on ui by setting env prefixed with 'SPARK_DRIVER_LOG_URL_'.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@sharkdtu do you mind:
1. Keep the github PR template: https://github.com/apache/spark/blob/master/.github/PULL_REQUEST_TEMPLATE
2. Fix the test failure: https://github.com/sharkdtu/spark/runs/2518937303?check_suite_focus=true

**Comment by sharkdtu** [comment]
@HyukjinKwon Thanks, PR description has been updated and the test failure has been fixed.

File: core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala [hunk 1]
```diff
@@ -655,6 +656,13 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     false
   }
 
+  override def getDriverLogUrls: Option[Map[String, String]] = {
+    val prefix = "SPARK_DRIVER_LOG_URL_"
+    val driverLogUrls = sys.env.filterKeys(_.startsWith(prefix))
+      .map(e => (e._1.substring(prefix.length).toLowerCase(Locale.ROOT), e._2)).toMap
```
Comment (gerashegalov) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala:662]: nit: I understand the code is just copied but this `toMap` seems unnecessary.

Comment (sharkdtu) [reply core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala:662]: it is for scala-2.13

Comment (gerashegalov) [reply core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala:662]: makes sense, thanks

**Comment by sharkdtu** [comment]
@HyukjinKwon Could you review this pr, and merge it if it is ok. Thanks a lot.

**Comment by fanyilun** [comment]
Does anyone follow up on this pr? It's quite useful for spark on k8s.

**Comment by pan3793** [comment]
I'm working on https://github.com/apache/spark/pull/38357, it covers this functionality.

**Pull request closed without merging**